fnet/network.go

```diff
@@ -26,3 +27,19 @@ func NormalizePort(port string) string {
 	}
 	return ":" + port
 }
+
+// GRPCDestination parses dest and returns dest:port based on dest type
+// being a hostname, IPv4 or IPv6 address.
+func GRPCDestination(dest, port string) string {
+	if ip := net.ParseIP(dest); ip != nil {
```

Comment:
I'm tempted to simplify this... and just again check for : (except if the : are inside [])
see the suggested additional tests below, I suggest to write the tests first and then change the code to pass

Reply:
Will do. The func will need to be updated. IMO it's best to pass in a single dest string that can be host, host:port or ip:port.